test/math_test.cpp

```diff
@@ -0,0 +1,31 @@
+#include "jakalib/math.h"
+
+using namespace jakalib;
+
+template <typename A, typename B>
+void assert_close(A a, B b) {
+  assert(std::abs(a - b) < 1e-10);
+}
+
+void hypotTest() {
+  assert_close(hypot(0.0, 0.0, 0.0), 0.0);
+  assert_close(hypot(3.0, 4.0, 0.0), 5.0);
+  assert_close(hypot(0.0, -3.0, 4.0), 5.0);
+  assert_close(hypot(-3.0, 0.0, 4.0), 5.0);
+  assert_close(hypot(2.0, -7.0, 26.0), 27.0);
+  assert_close(hypot(-0.2, 0.7, -2.6), 2.7);
+
+  long double x = std::numeric_limits<long double>::max() / 2.0;
```

Comment:
In order to properly check for loss of precision, consider using a large value with many significant digits. `max()` is platform specific, so best to divide by something like `2.12345678901234567890`